zap/src/main/java/org/zaproxy/zap/extension/ascan/filters/ScanFilter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascan.filters;
+
+import org.zaproxy.zap.model.StructuralNode;
+
+/**
+ * ScanFilter is called by the {@code HostProcess#canScanNode} to check if a given node is filtered
```

Comment:
This should be generic (e.g. "ScanFilter is called for each message to be scanned to check if it's filtered or not."), that's being called by `HostProcess#canScanNode` is an implementation detail which doesn't matter for implementations.

Reply:
done.